s3/s3.go

```diff
@@ -0,0 +1,250 @@
+package s3
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	awsS3 "github.com/aws/aws-sdk-go/service/s3"
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+)
+
+func init() {
+	extpoints.Register("s3", S3{})
+}
+
+type S3 struct{}
+
+func (S3) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (S3) Summary() string {
+	return "Uploads/downloads file to/from AWS S3."
+}
+
+func usage() string {
+	return `Upload/Download file to/from AWS S3.
+Usage:
+  taskcluster S3 put <file> <bucket> <prefix>
+  taskcluster S3 get <bucket> <prefix> [<target>]
+  taskcluster S3 help <subcommand>
+`
+}
+
+func (S3) Usage() string {
+	return usage()
+}
+
+func (S3) Execute(context extpoints.Context) bool {
+	argv := context.Arguments
+
+	// Check for command
+	command := argv["S3"].(string)
+	provider := extpoints.CommandProviders()[command]
+	if provider == nil {
+		panic(fmt.Sprintf("Unknown command: %s", command))
+	}
+
+	// Print help for subcommands
+	if argv["help"] == true {
+		subcommand, ok := argv["<subcommand>"].(string)
+		if !ok {
+			panic(fmt.Sprintf("Unknown subcommand: %s", subcommand))
+		}
+		if subcommand == "put" {
+			fmt.Println("Usage:\ntaskcluster S3 put <file> <bucket> <prefix>")
+		} else if subcommand == "get" {
+			fmt.Println("Usage:\ntaskcluster S3 get <bucket> <prefix> [<target>]")
+		} else {
+			fmt.Printf("Invalid subcommand.\n")
+		}
+		return true
+	}
+
+	// Parse for bucket and prefix
+	bucket, ok := argv["<bucket>"].(string)
+	if !ok {
```

Comment:
Required parameters are always present, or docopt will show an error message. No need to do the `ok` part.